clients/codelist/codelist.go

```diff
@@ -66,52 +67,57 @@ func (c *Client) Healthcheck() (string, error) {
 }
 
 // GetValues returns dimension values from the codelist api
-func (c *Client) GetValues(id string) (vals DimensionValues, err error) {
+func (c *Client) GetValues(ctx context.Context, serviceAuthToken string, id string) (DimensionValues, error) {
```

Comment:
Some function expect serviceAuthToken to be the last parameter whilst others like this one expect it to be after context or elsewhere in the list of parameters, might be worth us setting serviceAuthToken as the last parameter on:

1. GetValues
2. GetCodeListEditions
3. GetCodes
4. GetCodesByID
5. GetDatasetsByCode
6. doServiceRequest

Alternatively, we can have it always after context and change it in the other APIs to expected after context rather than auth tokens being the last parameters.

Reply:
I think each method should have params order `ctx, serviceAuthToken......`